### ISSUE TITLE
Inconsistency between setProperties and updateProperties with classes

### ISSUE THREAD

**Issue by kitsonk** [AFASSoftware/maquette#100]
When using `projector.merge()` or `dom.merge()`, maquette uses `setProperties()`.  `setProperties()` does not check for _unset_ classes, while `updateProperties()` does.

This can lead to some unexpected behaviour.  Given the following HTML:

```html
<div id="foo" class="foo bar"></div>
```

And the following code:

```ts
import { createProjector, h } from 'maquette';

function renderFunction () {
    return h('div', {
        classes: {
             foo: true,
             bar: false,
             baz: true
         }
    });
}

const projector = createProjector({});
projector.merge(document.getElementById('foo'), renderFunction);
```

You would get the following after the merge:

```html
<div id="foo" class="foo bar baz"></div>
```

If the render function at some point turns `bar` to `true` and then turns it `false`, `updateProperties` will start tracking it and properly remove it.

While I can understand maquette "ignoring" properties which aren't explicitly defined in the `VNode` (like `id` in this example), when they are expressly asserted it seems logical that maquette would ensure they are dealt with.  For example, if I specified `disabled: false` in the properties, maquette properly _removes_ this from the DOM node with `setProperties`.

`setProperties()` is only used by `initPropertiesAndChildren()`.  This is used both for `merge()` as `createDom()` and can understand that _unsetting_ classes on a newly created DOM node is excessive.  I don't know if a possible solution would be, when using merge, to use `updateProperties()` on the DOM node, but of course that would need some sort of _previous_ `VNode` to compare against.

**Comment by johan-gorter** [comment]
It looks like we could give setProperties an `assumeNew` parameter that disables unsetting classes.

**Comment by RickHoving** [comment]
@kitsonk do you think this should be included in [Maquette 3.0](https://dev.afas.nl/blog-dev/ideas-for-maquette-3)? If so we would like to have your oppinion on this isse #121
